[GCC 11] CocoaModel/src/Model.cc: error: 'this' pointer is null

Building CMSSW with gcc11 fails:
```
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/src/Model.cc: In static member function 'static void Model::readSystemDescription()':
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/src/Model.cc:437:28: error: 'this' pointer is null [-Werror=nonnull]
   437 |         meastemp->construct();
      |         ~~~~~~~~~~~~~~~~~~~^~
In file included from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/src/Model.cc:13:
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/interface/Measurement.h:35:8: note: in a call to non-static member function 'void Measurement::construct()'
   35 |   void construct();
      |        ^~~~~~~~~
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/src/Model.cc:440:28: error: 'this' pointer is null [-Werror=nonnull]
   440 |         meastemp->construct();
      |         ~~~~~~~~~~~~~~~~~~~^~
In file included from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/src/Model.cc:13:
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/Alignment/CocoaModel/interface/Measurement.h:35:8: note: in a call to non-static member function 'void Measurement::construct()'
   35 |   void construct();
      |        ^~~~~~~~~
```

Judging by [this line](https://github.com/cms-sw/cmssw/blob/master/Alignment/CocoaModel/src/Model.cc#L472), these two branches are not supposed to be executed. Looking at github history of this file, the commented out likes are there from the start of github era.

--- a/Alignment/CocoaModel/src/Model.cc
+++ b/Alignment/CocoaModel/src/Model.cc
@@ -19,8 +19,6 @@
 #include "Alignment/CocoaModel/interface/MeasurementDiffEntry.h"
 #include "Alignment/CocoaModel/interface/CocoaDaqReaderText.h"
 #include "Alignment/CocoaModel/interface/CocoaDaqReaderRoot.h"
-//t#include "Alignment/CocoaModel/interface/MeasurementDiffAngle.h"
-//t#include "Alignment/CocoaModel/interface/MeasurementCentreEntry.h"
 #include "Alignment/CocoaUtilities/interface/ALIUtils.h"
 #include "Alignment/CocoaModel/interface/EntryAngle.h"
 #include "Alignment/CocoaModel/interface/ParameterMgr.h"
@@ -431,12 +429,6 @@ void Model::readSystemDescription() {
       } else if (measType == ALIstring("COPS")) {
         meastemp = new MeasurementCOPS(4, measType, measName);
         meastemp->setConversionFactor(wordlist);
-        meastemp->construct();
-      } else if (measType == ALIstring("DIFFCENTRE")) {
-        //t       meastemp = new MeasurementDiffCentre( 1, measType, measName );
-        meastemp->construct();
-      } else if (measType == ALIstring("DIFFANGLE")) {
-        //t        meastemp = new MeasurementDiffAngle( 2, measType, measName );
         meastemp->construct();
       } else if (measType == ALIstring("DIFFENTRY")) {
         meastemp = new MeasurementDiffEntry(1, measType, measName);

